packages/provider-elasticsearch/src/utils/highlightUtil.js

```diff
@@ -0,0 +1,68 @@
+import _ from 'lodash/fp.js'
+import F from 'futil'
+
+let highlightContentRegEx = ({ pre_tags, post_tags }) =>
+  new RegExp(`(?<=${pre_tags[0]})(.*?)(?=${post_tags[0]})`, 'g')
+
+let getHighlightMatches = _.curry((nodeHighlight, highlight) =>
+  highlight.match(highlightContentRegEx(nodeHighlight))
+)
+
+let getWordListMatchesRegEx = (words) => new RegExp(words.join('|'), 'gi')
+
+let highlightWith = ({ pre_tags, post_tags }) =>
+  F.highlight(pre_tags[0], post_tags[0])
+
+let findHighlightTagsRegEx = ({ pre_tags, post_tags }) =>
+  new RegExp(`${pre_tags[0]}|${post_tags[0]}`, 'gi')
+
+let removeHighlightTags = (nodeHighlight, highlightString) =>
+  _.replace(findHighlightTagsRegEx(nodeHighlight), '', highlightString)
+
+let getFieldFromSubField = _.flow(
+  F.dotEncoder.decode,
+  _.dropRight(1),
+  F.dotEncoder.encode
+)
+
+export let mergeHitHighlights = (nodeHighlight, fields, hitHighlights) => {
```

Comment:
This generic fn should only take strings only, e.g. `mergeHitHighlights(hi1, hi2, ...)`